tests/osaka/eip7692_eof_v1/eip5450_stack/test_execution.py

```diff
@@ -0,0 +1,70 @@
+"""Test execution of EOF code in the context of the operand stack height."""
+
+import pytest
+
+from ethereum_test_exceptions import EOFException
+from ethereum_test_tools import Account, EOFStateTestFiller
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+from ethereum_test_types.eof.constants import MAX_RUNTIME_STACK_HEIGHT
+from ethereum_test_types.eof.v1 import Container, Section
+from ethereum_test_types.eof.v1.constants import (
+    MAX_CODE_INPUTS,
+    MAX_STACK_INCREASE_LIMIT,
+    NON_RETURNING_SECTION,
+)
+
+from .. import EOF_FORK_NAME
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-5450.md"
+REFERENCE_SPEC_VERSION = "f20b164b00ae5553f7536a6d7a83a0f254455e09"
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+@pytest.mark.parametrize("code_inputs", [0, 1, 16, 127, 128])
+@pytest.mark.parametrize("call_op", [Op.CALLF, Op.JUMPF])
+def test_execution_at_max_stack_height(
+    eof_state_test: EOFStateTestFiller, code_inputs: int, call_op: Op
+):
+    """
+    Test execution at the maximum runtime operand stack height (1024).
+    EOF doesn't allow to increase the stack height of a single code section more than 1023.
+    The effect of the maximum runtime stack height is achieved by using non-zero number
+    of the code section inputs and increasing the runtime stack to the limit accordingly.
+    The test pushes consecutive numbers starting from 0 (including inputs).
+    At the maximum stack height SSTORE is used so it should store 1022 at key 1023.
+    """
+    container = Container(
+        sections=[
+            Section.Code(
+                (
+                    sum(Op.PUSH1(x) for x in range(code_inputs))
+                    + call_op[1]
+                    + (Op.STOP if call_op == Op.CALLF else b"")
+                ),
+            ),
+            Section.Code(
+                sum(Op.PUSH2(x) for x in range(code_inputs, MAX_RUNTIME_STACK_HEIGHT))
+                + Op.SSTORE
+                + Op.POP * (MAX_RUNTIME_STACK_HEIGHT - Op.SSTORE.popped_stack_items)
+                + (Op.RETF if call_op == Op.CALLF else Op.STOP),
+                code_inputs=code_inputs,
+                code_outputs=0 if call_op == Op.CALLF else NON_RETURNING_SECTION,
+                max_stack_height=MAX_RUNTIME_STACK_HEIGHT,
+            ),
+        ],
+    )
+
+    exception = None
+    if code_inputs < (MAX_RUNTIME_STACK_HEIGHT - MAX_STACK_INCREASE_LIMIT):
```

Comment:
this still requires some looking at to figure out what's going on. What if you do
`if MAX_RUNTIME_STACK_HEIGHT - code_inputs > MAX_STACK_INCREASE_LIMIT`? you then make it somewhat easier to see that there is an implied stack increase equal to the left-hand side of the inequality.

Reply:
How about now?

Reply:
beautiful, thx!